examples/property_prediction/pretrain_gnns/chem/pretrain_masking.py

```diff
@@ -0,0 +1,214 @@
+                                          pred_edge[bg.find_edges(masked_edges_indices)[1]]
+                    loss_edge = criterion(masked_edges_logits, masked_edges_labels)
+                    edge_acc = compute_accuracy(masked_edges_logits, masked_edges_labels)
+                    loss = loss_node + loss_edge
+                else:
+                    loss = loss_node
+                    edge_acc = 0
+
+                tq_train.set_postfix_str(
+                    "Epoch: {} Step: {} Loss: {:.4f} node_acc: {:.4f} edge_acc: {:.4f}".format(epoch,
+                                                                                               step + 1,
+                                                                                               loss.item(),
+                                                                                               node_acc,
+                                                                                               edge_acc), refresh=False)
+
+                optimizer.zero_grad()
+                loss.backward()
+                optimizer.step()
+
+
+def main():
+    parser = argparse.ArgumentParser(description='pretrain_masking')
+    parser.add_argument('--device', type=int, default=0,
+                        help='which gpu to use if any (default: 0)')
+    parser.add_argument('--batch_size', type=int, default=256,
+                        help='input batch size for training (default: 256)')
+    parser.add_argument('--epochs', type=int, default=100,
+                        help='number of epochs to train (default: 100)')
+    parser.add_argument('--lr', type=float, default=0.001,
+                        help='learning rate (default: 0.001)')
+    parser.add_argument('--decay', type=float, default=0,
+                        help='weight decay (default: 0)')
+    parser.add_argument('--num_layer', type=int, default=5,
+                        help='number of GNN message passing layers (default: 5).')
+    parser.add_argument('--emb_dim', type=int, default=300,
+                        help='embedding dimensions (default: 300)')
+    parser.add_argument('--dropout_ratio', type=float, default=0,
+                        help='dropout ratio (default: 0)')
+    parser.add_argument('--mask_rate', type=float, default=0.15,
+                        help='dropout ratio (default: 0.15)')
+    parser.add_argument('--mask_edge', type=int, default=0,
+                        help='whether to mask edges or not together with atoms')
+    parser.add_argument('--JK', type=str, default="last",
+                        help='how the node features are combined across layers. last, sum, max or concat')
+    parser.add_argument('--dataset', type=str, default='zinc_standard_agent',
+                        help='root directory of dataset for pretraining')
+    parser.add_argument('--output_model_file', type=str, default='', help='filename to output the model')
```

Comment:
Maybe change "''" to something like "model.pth"?

Reply:
changed to "pretrain_masking.pth".